src/langchain/llm-chat.ts

```diff
@@ -3,117 +3,181 @@ import {
   BaseChatModelParams,
 } from '@langchain/core/language_models/chat_models';
 import {
+  AIMessage,
+  AIMessageChunk,
   BaseMessage,
-  MessageType,
-  SystemMessage,
 } from '@langchain/core/messages';
 import { CallbackManagerForLLMRun } from '@langchain/core/callbacks/manager';
-import { ChatResult } from '@langchain/core/outputs';
+import { ChatGenerationChunk, ChatResult } from '@langchain/core/outputs';
+import { BaseLanguageModelCallOptions } from '@langchain/core/language_models/base';
+import merge from 'lodash/merge.js';
 
-import { InvalidInputError } from '../errors.js';
-import { concatUnique } from '../helpers/common.js';
-import type { RequiredPartial } from '../helpers/types.js';
-import { TextGenerationCreateOutput } from '../schema.js';
-
-import { GenAIModel, GenAIModelOptions } from './llm.js';
+import { Client, Configuration } from '../client.js';
+import {
+  TextChatCreateInput,
+  TextChatCreateStreamInput,
+  TextGenerationCreateInput,
+} from '../schema.js';
+import { InternalError, InvalidInputError } from '../errors.js';
 
-export type RolesMapping = RequiredPartial<
-  Record<
-    MessageType,
-    {
-      stopSequence: string;
-    }
-  >,
-  'system'
->;
+export type GenAIChatModelParams = BaseChatModelParams &
+  Required<
+    Pick<TextGenerationCreateInput & TextChatCreateStreamInput, 'model_id'>
+  > &
+  Pick<
+    TextGenerationCreateInput & TextChatCreateStreamInput,
+    'prompt_id' | 'conversation_id' | 'parameters'
+  > & { configuration?: Configuration };
+export type GenAIChatModelOptions = BaseLanguageModelCallOptions &
+  Pick<GenAIChatModelParams, 'parameters'>;
 
-type Options = BaseChatModelParams &
-  GenAIModelOptions & {
-    rolesMapping: RolesMapping;
-  };
+export class GenAIChatModel extends BaseChatModel<GenAIChatModelOptions> {
+  protected readonly client: Client;
 
-export class GenAIChatModel extends BaseChatModel {
-  readonly #model: GenAIModel;
-  readonly #rolesMapping: RolesMapping;
+  public readonly modelId: GenAIChatModelParams['model_id'];
+  public readonly promptId: GenAIChatModelParams['prompt_id'];
+  public readonly conversationId: GenAIChatModelParams['conversation_id'];
+  protected parameters: GenAIChatModelParams['parameters'];
 
-  constructor(options: Options) {
+  constructor({
+    model_id,
+    prompt_id,
+    conversation_id,
+    parameters,
+    configuration,
+    ...options
+  }: GenAIChatModelParams) {
```

Comment:
Use javascript naming (`camelCase` and not `snake_case`). 
Could you ensure every parameter is set to the class and sent to the Client Text Generation?

Remove `GenAIChatModelParams` and just use the same one that is used for chat endpoint.

Reply:
> Use javascript naming (camelCase and not snake_case).

Snake case was chosen in the ctor for consistency with the parameters which are also snake cased.

> Could you ensure every parameter is set to the class and sent to the Client Text Generation?

They already are 🤔 

> Remove GenAIChatModelParams and just use the same one that is used for chat endpoint.

`GenAIChatModelParams` are mostly derived from chat endpoint types, however, there are slightly different requirements for some properties. There is also client configuration.

Reply:
>They already are 🤔
some are not, like `use_conversation_parameters`, `moderations` ...